### PR TITLE
Add request field

### PR DESCRIPTION
**Pull request by Lan-Jingbo**

### CODE REVIEW

File: src/main/java/seedu/guest/model/guest/Guest.java [hunk 1]
```diff
@@ -20,20 +20,22 @@ public class Guest {
...
     private final Bill bill;
+    private final Request request;
 
     /**
      * Every field must be present and not null.
      */
     public Guest(Name name, Phone phone, Email email, DateRange dateRange,
-                 NumberOfGuests numberOfGuests, IsRoomClean isRoomClean, Bill bill) {
-        requireAllNonNull(name, phone, email, dateRange, numberOfGuests, isRoomClean, bill);
+                 NumberOfGuests numberOfGuests, IsRoomClean isRoomClean, Bill bill, Request requests) {
```
Comment (kangzongxian) [inline src/main/java/seedu/guest/model/guest/Guest.java]: Should this requests be request since we use request in other parts of the program?

File: src/main/java/seedu/guest/model/guest/Request.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package seedu.guest.model.guest;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.guest.commons.util.AppUtil.checkArgument;
+
+/**
+ * The class Requests.
```
Comment (kangzongxian) [inline src/main/java/seedu/guest/model/guest/Request.java]: Should this be Request? Also, should we follow the same documentation-style for the other field classes as well (e.g Address, Name, etc...)?

File: src/main/java/seedu/guest/model/guest/Request.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+     */
+    public Request() {
+        value = "Nothing";
+    }
+
+    /**
+     * The constructor with specific request.
+     * @param str the specific request
+     */
+    public Request(String str) {
```
Comment (kangzongxian) [inline src/main/java/seedu/guest/model/guest/Request.java]: Should we follow the same documentation-style for the other field classes as well for the constructor?

File: src/main/java/seedu/guest/model/guest/Request.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        requireNonNull(str);
+        checkArgument(isValidRequest(str), MESSAGE_CONSTRAINTS);
+        value = str;
+    }
+
+    /**
+     * To check whether the request is valid.
+     * @param str
```
Comment (kangzongxian) [inline src/main/java/seedu/guest/model/guest/Request.java]: Should we follow the same documentation-style for the other field classes as well?

File: src/main/java/seedu/guest/model/guest/Request.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+     * @return boolean
+     */
+    public static boolean isValidRequest(String str) {
+        return (!str.equals(""));
+    }
+
+    /**
+     * return the String representation of request.
+     * @return
```
Comment (kangzongxian) [inline src/main/java/seedu/guest/model/guest/Request.java]: Should we not add documentation for this to make it more in line with the other field classes?

File: src/main/java/seedu/guest/model/guest/Request.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    public String toString() {
+        return value;
+    }
+
+    /**
+     * to check whether this request is the same as another.
+     * @param other other object, might be requeest
+     * @return boolean
+     */
+    @Override
```
Comment (kangzongxian) [inline src/main/java/seedu/guest/model/guest/Request.java:50]: Should we not add documentation for this to make it more in line with the other field classes?

File: src/main/java/seedu/guest/model/guest/Request.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Request // instanceof handles nulls
+                && value.equals(((Request) other).value)); // state check
+    }
+
+    /**
+     * To override hashcode.
+     * @return integer
+     */
```
Comment (kangzongxian) [inline src/main/java/seedu/guest/model/guest/Request.java]: Should we not add documentation for this to make it more in line with the other field classes?

File: src/main/java/seedu/guest/ui/GuestCard.java [hunk 8]
```diff
@@ -41,6 +41,8 @@ public class GuestCard extends UiPart<Region> {
     private Label isRoomClean;
     @FXML
     private Label bill;
+    @FXML
+    private Label requests;
```
Comment (kangzongxian) [inline src/main/java/seedu/guest/ui/GuestCard.java]: Should this be request instead of requests?

File: src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json [hunk 9]
```diff
@@ -7,54 +7,61 @@
...
   }, {
     "name": "Carl Kurz",
     "phone": "95352563",
     "email": "heinz@example.com",
     "dateRange": "30/12/22 - 02/01/23",
     "numberOfGuests": "3",
     "isRoomClean": "no",
-    "bill": "19.99"
+    "bill": "19.99",
+    "request" : "I wanna sleep"
```
Comment (kangzongxian) [inline src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json]: Should we have more probable request examples since it represents a typical GuestBook?

File: src/main/java/seedu/guest/model/guest/Request.java (hunk 3 above)
Comment (kangzongxian) [inline src/main/java/seedu/guest/model/guest/Request.java]: Should we name the parameter with a more contextual naming rather than str?

File: src/test/java/seedu/guest/logic/parser/ParserUtilTest.java [hunk 10]
```diff
@@ -32,6 +34,7 @@ public class ParserUtilTest {
     private static final String VALID_NUMBER_OF_GUESTS = "4";
     private static final String VALID_IS_ROOM_CLEAN = "yes";
     private static final String VALID_BILL = "10.00";
+    private static final String VALID_REQUEST = "I wanna book for room service";
```
Comment (kangzongxian) [inline src/test/java/seedu/guest/logic/parser/ParserUtilTest.java]: Should we avoid slangs in our string to make it more proper?

File: src/test/java/seedu/guest/model/util/SampleDataUtilTest.java [hunk 11]
```diff
@@ -54,22 +55,22 @@ public void getSampleGuestBook() {
         GuestBook exceptedGb = new GuestBook();
         exceptedGb.addGuest(new Guest(new Name("James Oliver"), new Phone("98345344"),
                 new Email("jamesoliver@example.com"), new DateRange("15/09/22 - 19/09/22"),
-                new NumberOfGuests("2"), new IsRoomClean("no"), new Bill()));
+                new NumberOfGuests("2"), new IsRoomClean("no"), new Bill(), new Request()));
         exceptedGb.addGuest(new Guest(new Name("Kayla Smith"), new Phone("91823471"),
                 new Email("kaylasmith@example.com"), new DateRange("22/01/23 - 25/01/23"),
-                new NumberOfGuests("3"), new IsRoomClean("no"), new Bill("10")));
+                new NumberOfGuests("3"), new IsRoomClean("no"), new Bill("10"), new Request("hi")));
```
Comment (kangzongxian) [inline src/test/java/seedu/guest/model/util/SampleDataUtilTest.java]: Should we have more appropriate examples for passing in the string value to construct a Request?

File: src/test/java/seedu/guest/storage/JsonSerializableGuestBookTest.java [hunk 12]
```diff
@@ -26,7 +25,7 @@ public void toModelType_typicalGuestsFile_success() throws Exception {
                 JsonSerializableGuestBook.class).get();
         GuestBook guestBookFromFile = dataFromFile.toModelType();
         GuestBook typicalPersonsGuestBook = TypicalGuests.getTypicalGuestBook();
-        assertEquals(guestBookFromFile, typicalPersonsGuestBook);
+        assert(guestBookFromFile.toString().equals(typicalPersonsGuestBook.toString()));
```
Comment (kangzongxian) [inline src/test/java/seedu/guest/storage/JsonSerializableGuestBookTest.java]: Should we change this part?

File: src/test/java/seedu/guest/testutil/TypicalGuests.java [hunk 13]
```diff
@@ -27,42 +27,42 @@ public class TypicalGuests {
...
             .build();
     public static final Guest DANIEL = new GuestBuilder().withName("Daniel Meier").withPhone("87652533")
             .withEmail("cornelia@example.com").withDateRange("29/05/23 - 03/06/23")
-            .withNumberOfGuests("4").withIsRoomClean("yes").withBill("50")
+            .withNumberOfGuests("4").withIsRoomClean("yes").withBill("50").withRequest("Room service")
             .build();
     public static final Guest ELLE = new GuestBuilder().withName("Elle Meyer").withPhone("9482224")
             .withEmail("werner@example.com").withDateRange("08/04/23 - 22/04/23")
-            .withNumberOfGuests("1").withIsRoomClean("no").withBill("100")
+            .withNumberOfGuests("1").withIsRoomClean("no").withBill("100").withRequest("Room service")
```
Comment (kangzongxian) [inline src/test/java/seedu/guest/testutil/TypicalGuests.java:52]: Should we have more varied requests for typical guests?

**Review by kangzongxian — CHANGES_REQUESTED** [review]

File: src/test/java/seedu/guest/testutil/TypicalGuests.java (hunk 13 above)
Comment (Lan-Jingbo) [reply src/test/java/seedu/guest/testutil/TypicalGuests.java:52]: fixed all things

File: src/main/java/seedu/guest/logic/commands/AddCommand.java [hunk 14]
```diff
@@ -29,14 +30,16 @@ public class AddCommand extends Command {
             + PREFIX_DATE_RANGE + "DATE_RANGE "
             + PREFIX_NUMBER_OF_GUESTS + "NUMBER_OF_GUESTS "
             + PREFIX_IS_ROOM_CLEAN + "IS_ROOM_CLEAN "
+            + PREFIX_REQUEST + "REQUEST(Optional) "
```
Comment (jehousoh) [inline src/main/java/seedu/guest/logic/commands/AddCommand.java]: Should we use square brackets to indicate optional? eg. "[" + PREFIX_REQUEST + "REQUEST]...\n" (This was how 'tag' was done in AB3).

File: src/main/java/seedu/guest/logic/commands/AddCommand.java [hunk 15]
```diff
@@ -29,14 +30,16 @@ public class AddCommand extends Command {
...
             + "Example: " + COMMAND_WORD + " "
             + PREFIX_NAME + "John Doe "
             + PREFIX_PHONE + "98765432 "
             + PREFIX_EMAIL + "johnd@example.com "
             + PREFIX_ROOM + "05-73 "
             + PREFIX_DATE_RANGE + "13/09/22 - 15/09/22 "
             + PREFIX_NUMBER_OF_GUESTS + "1 "
-            + PREFIX_IS_ROOM_CLEAN + "yes";
+            + PREFIX_IS_ROOM_CLEAN + "yes"
+            + "(Optional: " + PREFIX_REQUEST + "Apply for room service" + ")";
```
Comment (jehousoh) [inline src/main/java/seedu/guest/logic/commands/AddCommand.java]: Should we remove the Optional: and brackets here to just put the request? Also "Apply for room service" is a weird request should we change it to something more normal?

File: src/main/java/seedu/guest/model/guest/Guest.java [hunk 16]
```diff
@@ -21,12 +21,13 @@ public class Guest {
...
     private final Bill bill;
+    private final Request request;
 
     /**
      * Every field must be present and not null.
      */
     public Guest(Name name, Phone phone, Email email, Room room, DateRange dateRange,
-                 NumberOfGuests numberOfGuests, IsRoomClean isRoomClean, Bill bill) {
+                 NumberOfGuests numberOfGuests, IsRoomClean isRoomClean, Bill bill, Request request) {
         requireAllNonNull(name, phone, email, dateRange, numberOfGuests, isRoomClean, bill);
```
Comment (jehousoh) [inline src/main/java/seedu/guest/model/guest/Guest.java]: Should request be here as well?

File: src/main/java/seedu/guest/model/guest/Request.java [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+ */
+public class Request {
+    public static final String MESSAGE_CONSTRAINTS = "The request should not be null";
+    public final String value;
+
+    /**
+     * The constructor with no specific request.
+     */
+    public Request() {
+        value = "Nothing";
```
Comment (jehousoh) [inline src/main/java/seedu/guest/model/guest/Request.java]: Should our default be "-" instead?

File: src/main/java/seedu/guest/model/guest/Request.java [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+        value = str;
+    }
+
+    /**
+     * To check whether the Request is valid.
+     * @param inputRequest
+     * @return boolean
+     */
+    public static boolean isValidRequest(String inputRequest) {
+        return (!inputRequest.equals(""));
```
Comment (jehousoh) [inline src/main/java/seedu/guest/model/guest/Request.java]: Why do we not allow ""?

File: src/main/java/seedu/guest/logic/parser/AddCommandParser.java [hunk 19]
```diff
@@ -36,26 +38,43 @@ public class AddCommandParser implements Parser<AddCommand> {
...
-        DateRange dateRange = ParserUtil.parseDateRange(argMultimap.getValue(PREFIX_DATE_RANGE).get());
-        NumberOfGuests numberOfGuests = ParserUtil
-                .parseNumberOfGuests(argMultimap.getValue(PREFIX_NUMBER_OF_GUESTS).get());
-        IsRoomClean isRoomClean = ParserUtil
-                .parseIsRoomClean(argMultimap.getValue(PREFIX_IS_ROOM_CLEAN).get());
-        Bill bill = new Bill();
-        Guest guest = new Guest(name, phone, email, room, dateRange, numberOfGuests, isRoomClean, bill);
-        return new AddCommand(guest);
+        if (arePrefixesPresent(argMultimap, PREFIX_REQUEST)) {
+            Request request = ParserUtil.parseRequest(argMultimap.getValue(PREFIX_REQUEST).get());
```
Comment (jehousoh) [inline src/main/java/seedu/guest/logic/parser/AddCommandParser.java]: Should this line be after the Bill line for consistency?

File: src/main/java/seedu/guest/model/util/SampleDataUtil.java [hunk 20]
```diff
@@ -20,22 +21,22 @@ public static Guest[] getSampleGuests() {
...
-                    new IsRoomClean("yes"), new Bill("50.50")),
+                    new IsRoomClean("yes"), new Bill("50.50"), new Request("Room service")),
             new Guest(new Name("Irfan Ibrahim"), new Phone("92492021"), new Email("irfan@example.com"),
                     new Room("09-72"), new DateRange("29/05/23 - 03/06/23"), new NumberOfGuests("1"),
-                    new IsRoomClean("no"), new Bill("100")),
+                    new IsRoomClean("no"), new Bill("100"), new Request("Room service")),
             new Guest(new Name("Roy Balakrishnan"), new Phone("92624417"), new Email("royb@example.com"),
                     new Room("04-23"), new DateRange("30/12/22 - 02/01/23"), new NumberOfGuests("4"),
-                    new IsRoomClean("yes"), new Bill("1000.01"))
+                    new IsRoomClean("yes"), new Bill("1000.01"), new Request("Room service"))
```
Comment (jehousoh) [inline src/main/java/seedu/guest/model/util/SampleDataUtil.java]: Should we have different Requests for each guests instead of mostly "Room service"?

File: src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json [hunk 21]
```diff
@@ -17,7 +18,8 @@
     "dateRange": "30/09/22 - 02/10/22",
     "numberOfGuests": "2",
     "isRoomClean": "yes",
-    "bill": "1.00"
+    "bill": "1.00",
+    "request" : "I want to clean my room for one more time"
```
Comment (jehousoh) [inline src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json]: Should it be "I want my room to be cleaned again" for proper grammar?

File: src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json [hunk 22]
```diff
@@ -26,7 +28,8 @@
     "dateRange": "30/12/22 - 02/01/23",
     "numberOfGuests": "3",
     "isRoomClean": "no",
-    "bill": "19.99"
+    "bill": "19.99",
+    "request" : "There are too many insects in my room"
```
Comment (jehousoh) [inline src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json]: Should it be phrased as a request instead of a comment?

File: src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json [hunk 23]
```diff
@@ -53,7 +58,8 @@
     "dateRange": "21/10/22 - 22/10/22",
     "numberOfGuests": "2",
     "isRoomClean": "no",
-    "bill": "999.99"
+    "bill": "999.99",
+    "request" : "The neighbour is too noisy, could you help me shoot him or her?"
```
Comment (jehousoh) [inline src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json]: Should this be more normal?

File: src/test/java/seedu/guest/logic/commands/CommandTestUtil.java [hunk 24]
```diff
@@ -102,8 +102,6 @@ public static void assertCommandSuccess(Command command, Model actualModel, Comm
             Model expectedModel) {
         try {
             CommandResult result = command.execute(actualModel);
-            assertEquals(expectedCommandResult, result);
-            assertEquals(expectedModel, actualModel);
```
Comment (jehousoh) [inline src/test/java/seedu/guest/logic/commands/CommandTestUtil.java]: Why were these removed?

File: src/test/java/seedu/guest/logic/parser/ParserUtilTest.java [hunk 25]
```diff
@@ -25,6 +26,7 @@ public class ParserUtilTest {
     private static final String INVALID_DATE_RANGE = "13/09/22 - 13/09/22";
     private static final String INVALID_NUMBER_OF_GUESTS = "5";
     private static final String INVALID_IS_ROOM_CLEAN = "false";
+    private static final String INVALID_REQUEST = " ";
```
Comment (jehousoh) [inline src/test/java/seedu/guest/logic/parser/ParserUtilTest.java]: Why is this invalid?

File: src/test/java/seedu/guest/logic/parser/ParserUtilTest.java [hunk 26]
```diff
@@ -35,6 +37,7 @@ public class ParserUtilTest {
     private static final String VALID_NUMBER_OF_GUESTS = "4";
     private static final String VALID_IS_ROOM_CLEAN = "yes";
     private static final String VALID_BILL = "10.00";
+    private static final String VALID_REQUEST = "I want to book for room service";
```
Comment (jehousoh) [inline src/test/java/seedu/guest/logic/parser/ParserUtilTest.java]: Should we change it to "May I have room service?"

File: src/test/java/seedu/guest/model/util/SampleDataUtilTest.java [hunk 27]
```diff
@@ -57,27 +63,32 @@ public void getSampleGuestBook() {
...
                 new DateRange("28/04/23 - 29/04/23"),
-                new NumberOfGuests("2"), new IsRoomClean("no"), new Bill("50")));
+                new NumberOfGuests("2"), new IsRoomClean("no"), new Bill("50"),
+                new Request("My bed is not comfortable")));
         exceptedGb.addGuest(new Guest(new Name("Ibrahim Irfan"), new Phone("84912293"),
                 new Email("irfan@example.com"), new Room("06-33"),
                 new DateRange("23/05/23 - 03/06/23"),
-                new NumberOfGuests("1"), new IsRoomClean("yes"), new Bill("999.99")));
+                new NumberOfGuests("1"), new IsRoomClean("yes"), new Bill("999.99"),
+                new Request("Could you help me to do the team project?")));
```
Comment (jehousoh) [inline src/test/java/seedu/guest/model/util/SampleDataUtilTest.java]: Should we make this more related to the hotel?

File: src/test/java/seedu/guest/model/util/SampleDataUtilTest.java [hunk 28]
```diff
@@ -57,27 +63,32 @@ public void getSampleGuestBook() {
...
         exceptedGb.addGuest(new Guest(new Name("Dave Balakrishnan"), new Phone("92821374"),
                 new Email("daveb@example.com"), new Room("07-55"),
                 new DateRange("30/12/22 - 02/01/23"),
-                new NumberOfGuests("3"), new IsRoomClean("no"), new Bill("1000.01")));
+                new NumberOfGuests("3"), new IsRoomClean("no"), new Bill("1000.01"),
+                new Request("Could your officer help me complete the assignments, it is too hard.")));
```
Comment (jehousoh) [inline src/test/java/seedu/guest/model/util/SampleDataUtilTest.java]: Should we make this more related to the hotel?

File: src/test/java/seedu/guest/testutil/GuestBuilder.java [hunk 29]
```diff
@@ -22,6 +23,7 @@ public class GuestBuilder {
     public static final String DEFAULT_NUMBER_OF_GUESTS = "1";
     public static final String DEFAULT_IS_ROOM_CLEAN = "yes";
     public static final String DEFAULT_BILL = "0.00";
+    public static final String DEFAULT_REQUEST = "Nothing";
```
Comment (jehousoh) [inline src/test/java/seedu/guest/testutil/GuestBuilder.java]: Should we change the default to "-"?

File: src/test/java/seedu/guest/testutil/TypicalGuests.java [hunk 30]
```diff
@@ -30,41 +30,47 @@ public class TypicalGuests {
...
             .build();
     public static final Guest ELLE = new GuestBuilder().withName("Elle Meyer").withPhone("9482224")
             .withEmail("werner@example.com").withRoom("03-25").withDateRange("08/04/23 - 22/04/23")
-            .withNumberOfGuests("1").withIsRoomClean("no").withBill("100")
+            .withNumberOfGuests("1").withIsRoomClean("no").withBill("100").withRequest("Room service")
             .build();
     public static final Guest FIONA = new GuestBuilder().withName("Fiona Kunz").withPhone("9482427")
             .withEmail("lydia@example.com").withRoom("06-22").withDateRange("21/10/22 - 22/10/22")
             .withNumberOfGuests("2").withIsRoomClean("no").withBill("999.99")
+            .withRequest("The neighbour is too noisy, could you help me shoot him or her?")
```
Comment (jehousoh) [inline src/test/java/seedu/guest/testutil/TypicalGuests.java]: Likewise, should this be more normal?

File: src/test/java/seedu/guest/testutil/TypicalGuests.java [hunk 31]
```diff
@@ -30,41 +30,47 @@ public class TypicalGuests {
...
     // Manually added
     public static final Guest HOON = new GuestBuilder().withName("Hoon Meier").withPhone("8482424")
             .withEmail("stefan@example.com").withRoom("02-21").withDateRange("01/05/23 - 01/01/24")
-            .withNumberOfGuests("1").withIsRoomClean("no").withBill("6969.69")
+            .withNumberOfGuests("1").withIsRoomClean("no").withBill("6969.69").withRequest("Room service")
             .build();
     public static final Guest IDA = new GuestBuilder().withName("Ida Mueller").withPhone("8482131")
             .withEmail("hans@example.com").withRoom("05-36").withDateRange("11/11/23 - 12/11/23")
-            .withNumberOfGuests("4").withIsRoomClean("yes").withBill("9999999999")
+            .withNumberOfGuests("4").withIsRoomClean("yes").withBill("9999999999").withRequest("Room service")
```
Comment (jehousoh) [inline src/test/java/seedu/guest/testutil/TypicalGuests.java]: Should we have a different request compared to HOON?

**Review by jehousoh — CHANGES_REQUESTED** [review]
Good work on the commands but please take note of your example requests!

File: src/main/java/seedu/guest/model/guest/Request.java (hunk 18 above)
Comment (Lan-Jingbo) [reply src/main/java/seedu/guest/model/guest/Request.java]: cuz it will be null. My thoughts is that either there are no request or there must be at least a word

File: src/main/java/seedu/guest/model/guest/Request.java [hunk 32]
```diff
@@ -0,0 +1,68 @@
+package seedu.guest.model.guest;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.guest.commons.util.AppUtil.checkArgument;
+
+/**
+ * The class Request.
```
Comment (kangzongxian) [inline src/main/java/seedu/guest/model/guest/Request.java]: Should we follow the same documentation style for the other field classes? (E.g Represents..., Guarantees...)

File: src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json [hunk 33]
```diff
@@ -62,6 +68,7 @@
     "dateRange": "01/01/23 - 07/01/23",
     "numberOfGuests": "3",
     "isRoomClean": "yes",
-    "bill": "1000.01"
+    "bill": "1000.01",
+    "request" : "Help me to fix Wi-Fi!"
```
Comment (kangzongxian) [inline src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json]: Should this be phrased in the perspective of a hotel manager? E.g "Fix Wifi", as this will be entered by the hotel manager him/herself

File: src/test/java/seedu/guest/logic/commands/EditCommandTest.java [hunk 34]
```diff
@@ -35,7 +35,7 @@ public class EditCommandTest {
 
     @Test
     public void execute_allFieldsSpecifiedUnfilteredList_success() {
-        Guest editedGuest = new GuestBuilder().build();
+        Guest editedGuest = new GuestBuilder().build(); //Amy Bee
```
Comment (kangzongxian) [inline src/test/java/seedu/guest/logic/commands/EditCommandTest.java]: Should there be an "Amy Bee" comment here?

**Review by kangzongxian — CHANGES_REQUESTED** [review]

**Review by kangzongxian — APPROVED** [review]
Looks good to me!

File: src/main/java/seedu/guest/model/guest/Guest.java (hunk 16 above)
Comment (Lan-Jingbo) [reply src/main/java/seedu/guest/model/guest/Guest.java]: Should, because it need to mark whether the guest have request
if no request, I set as "-"

File: src/test/java/seedu/guest/logic/commands/EditCommandTest.java (hunk 34 above)
Comment (Lan-Jingbo) [reply src/test/java/seedu/guest/logic/commands/EditCommandTest.java]: changed

File: src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json (hunk 33 above)
Comment (Lan-Jingbo) [reply src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json]: changed

File: src/main/java/seedu/guest/model/guest/Request.java (hunk 32 above)
Comment (Lan-Jingbo) [reply src/main/java/seedu/guest/model/guest/Request.java]: changed

File: src/test/java/seedu/guest/testutil/TypicalGuests.java (hunk 31 above)
Comment (Lan-Jingbo) [reply src/test/java/seedu/guest/testutil/TypicalGuests.java]: changed

File: src/test/java/seedu/guest/testutil/TypicalGuests.java (hunk 30 above)
Comment (Lan-Jingbo) [reply src/test/java/seedu/guest/testutil/TypicalGuests.java]: changed

File: src/test/java/seedu/guest/testutil/GuestBuilder.java (hunk 29 above)
Comment (Lan-Jingbo) [reply src/test/java/seedu/guest/testutil/GuestBuilder.java]: changed

File: src/test/java/seedu/guest/model/util/SampleDataUtilTest.java (hunk 28 above)
Comment (Lan-Jingbo) [reply src/test/java/seedu/guest/model/util/SampleDataUtilTest.java]: changed

File: src/test/java/seedu/guest/model/util/SampleDataUtilTest.java (hunk 27 above)
Comment (Lan-Jingbo) [reply src/test/java/seedu/guest/model/util/SampleDataUtilTest.java]: changed

File: src/test/java/seedu/guest/logic/parser/ParserUtilTest.java (hunk 26 above)
Comment (Lan-Jingbo) [reply src/test/java/seedu/guest/logic/parser/ParserUtilTest.java]: changed

File: src/test/java/seedu/guest/logic/parser/ParserUtilTest.java (hunk 25 above)
Comment (Lan-Jingbo) [reply src/test/java/seedu/guest/logic/parser/ParserUtilTest.java]: changed

File: src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json (hunk 23 above)
Comment (Lan-Jingbo) [reply src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json]: changed

File: src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json (hunk 22 above)
Comment (Lan-Jingbo) [reply src/test/data/JsonSerializableGuestBookTest/typicalGuestGuestBook.json]: changed

File: src/main/java/seedu/guest/model/util/SampleDataUtil.java (hunk 20 above)
Comment (Lan-Jingbo) [reply src/main/java/seedu/guest/model/util/SampleDataUtil.java]: changed

File: src/main/java/seedu/guest/logic/parser/AddCommandParser.java (hunk 19 above)
Comment (Lan-Jingbo) [reply src/main/java/seedu/guest/logic/parser/AddCommandParser.java]: changed

File: src/main/java/seedu/guest/model/guest/Request.java (hunk 17 above)
Comment (Lan-Jingbo) [reply src/main/java/seedu/guest/model/guest/Request.java]: changed

File: src/main/java/seedu/guest/logic/commands/AddCommand.java (hunk 14 above)
Comment (Lan-Jingbo) [reply src/main/java/seedu/guest/logic/commands/AddCommand.java]: changed

File: src/test/java/seedu/guest/logic/commands/CommandTestUtil.java (hunk 24 above)
Comment (Lan-Jingbo) [reply src/test/java/seedu/guest/logic/commands/CommandTestUtil.java]: fixed

File: src/main/java/seedu/guest/model/guest/Guest.java (hunk 1 above)
Comment (Lan-Jingbo) [reply src/main/java/seedu/guest/model/guest/Guest.java]: changed

File: src/main/java/seedu/guest/model/guest/Request.java (hunk 2 above)
Comment (Lan-Jingbo) [reply src/main/java/seedu/guest/model/guest/Request.java]: changed

File: src/main/java/seedu/guest/model/guest/Request.java (hunk 3 above)
Comment (Lan-Jingbo) [reply src/main/java/seedu/guest/model/guest/Request.java]: changed

File: src/main/java/seedu/guest/model/guest/Request.java (hunk 4 above)
Comment (Lan-Jingbo) [reply src/main/java/seedu/guest/model/guest/Request.java]: changed

File: src/main/java/seedu/guest/model/guest/Request.java (hunk 5 above)
Comment (Lan-Jingbo) [reply src/main/java/seedu/guest/model/guest/Request.java]: changed

**Review by jehousoh — APPROVED** [review]
lgtm
